src/components/accessWrappers/EnrollmentRoute.tsx

```diff
@@ -5,21 +5,34 @@ import NotFound from '../pages/NotFound';
 
 import { EnrollmentPermissions } from '@/modules/permissions/types';
 import { useHasPermissions } from '@/modules/permissions/useHasPermissionsHooks';
+import { DataCollectionFeatureRole } from '@/types/gqlTypes';
 import { generateSafePath } from '@/utils/pathEncoding';
 
 /**
  * Permission wrapper to be used for outlets of the EnrollmentDashboard
  */
 const EnrollmentRoute: React.FC<
   React.PropsWithChildren<{
-    permissions: EnrollmentPermissions | EnrollmentPermissions[];
+    permissions?: EnrollmentPermissions | EnrollmentPermissions[];
     redirectRoute?: string;
+    dataCollectionFeature?: DataCollectionFeatureRole;
   }>
-> = ({ permissions, redirectRoute, children }) => {
+> = ({ permissions, redirectRoute, dataCollectionFeature, children }) => {
   // Use dashboard outlet context that gets set in EnrollmentDashboard
-  const { enrollment } = useEnrollmentDashboardContext();
-  const hasPermission = useHasPermissions(enrollment?.access, permissions);
-  if (!hasPermission) {
+  const { enrollment, enabledFeatures } = useEnrollmentDashboardContext();
+  const hasPermission = useHasPermissions(
+    enrollment?.access,
+    permissions || []
+  );
+
+  if (
+    dataCollectionFeature &&
+    !enabledFeatures.includes(dataCollectionFeature)
+  ) {
+    return <NotFound />;
+  }
+
+  if (!!permissions && !hasPermission) {
```

Comment:
Is it intentional that this will always return not found if you pass `permissions=[]`? We could make it more robust to handle that in the same way:
`if (ensureArray(permissions).length > 0 && !hasPermission) {`

Reply:
I agree with your suggestion and made the change to make it more robust. With this, we could also now revert the change I mentioned [here](https://github.com/greenriver/hmis-frontend/pull/959#discussion_r1804664499) making `permissions` an optional param. But I kind of like that change anyway, it seems like a clearer api for the Route components now that they are not only used for checking permissions